hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/rollback/BaseRollbackActionExecutor.java

```diff
@@ -88,11 +88,11 @@ public BaseRollbackActionExecutor(HoodieEngineContext context,
     this.resolvedInstant = instantToRollback;
     this.deleteInstants = deleteInstants;
     this.skipTimelinePublish = skipTimelinePublish;
-    this.useMarkerBasedStrategy = useMarkerBasedStrategy;
-    if (useMarkerBasedStrategy) {
-      ValidationUtils.checkArgument(!instantToRollback.isCompleted(),
-          "Cannot use marker based rollback strategy on completed instant:" + instantToRollback);
+    if (instantToRollback.isCompleted() && useMarkerBasedStrategy) {
+      useMarkerBasedStrategy = false;
+      LOG.warn("Cannot use marker based rollback strategy on completed instant:" + instantToRollback + ", the strategy has been automatically disabled.");
```

Comment:
this is a behavior change. @yihua do you know why this validation in the first place?

Reply:
@watermelon12138 this does not look like a bug, can you change the jira type from bug to improvement? And let's hold off this PR before @yihua confirms on the validation scenario.

Reply:
@xushiyan @yihua ok，i see. I have changed the jira type from bug to improvement. And the reason why CI fail is that some uts expect IllegalArgumentException during the rollback of the completed commit. Is that normal? why can't we just roll back completed instant?

Reply:
The validation is required because the marker-based rollback cannot be applied to a complete instant since there is no marker for complete instants.

Instead of fixing the logic here to hide the anti-usage pattern, the caller of `BaseRollbackActionExecutor` should make sure that marker-based rollback is not used for completed instants.  @watermelon12138 could you clarify which UTs are relevant and why we need the fix here?

Reply:
Ok this means we should not make this change here; it is against the original design.